docs/Debugging.md

```diff
@@ -0,0 +1,81 @@
+Debugging
+=========
+
+SGX-LKL provides a wrapper around gdb called `sgx-lkl-gdb`.
+This wrapper is compatible with IDEs and can be used instead of `gdb`.
+`sgx-lkl-gdb` automatically loads the SGX-LKL gdb plugin which
+ensures that debug symbols (if available) are loaded correctly. 
+When running in hardware mode, `sgx-lkl-gdb` uses the corresponding SGX debug
+instructions to read from and write to enclave memory.
+
+## Prerequisites
+
+### Open Enclave
+
+`sgx-lkl-gdb` relies on the Open Enclave gdb plugin.
+[Install Open Enclave](https://github.com/openenclave/openenclave/blob/master/docs/GettingStartedDocs/install_oe_sdk-Ubuntu_18.04.md) if necessary before continuing.
+
+The instructions below assume that Open Enclave was installed in the default directory `/opt/openenclave`.
+If you used a different directory, run `. <OE_PREFIX>/share/openenclave/openenclaverc` first.
+
+### Build mode
+
+Debugging enclave code (SGX-LKL as well as the application code) is only possible in debug and non-release builds of SGX-LKL.
+
+### Encrypted disks
+
+TODO add note on how to handle situations when disk keys are retrieved from a remote service that would not release them to debug/non-release builds of SGX-LKL (either re-create disks unencrypted, or, if available locally, hard-code key in app config for local debugging)
+
+### Confidential Containers
+
+The instructions below assume that disk images and configuration are available outside of a Confidential Container. Debugging of Confidential Containers (via Docker) is not supported currently.
+
+## Option A: Debugging with an SGX-LKL installation
+
+The instructions below assume that SGX-LKL was installed in `/opt/sgx-lkl`.
+
+To debug an application, invoke `sgx-lkl-gdb` as follows:
+```
+/opt/sgx-lkl/bin/sgx-lkl-gdb --args /opt/sgx-lkl/bin/sgx-lkl-run-oe --hw-debug root_disk.img /app
```

Comment:
This is not accurate enough. It should be `<path-to-app>`, plus optionally parameters to `app`

Reply:
The general idea is to prefix `sgx-lkl-gdb --args` to whatever command we used to launch the application in sgx-lkl.

Reply:
I changed the text and made it clear that the command is just an example.